modular/gater/object_handler.go

```diff
@@ -519,8 +504,29 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
 	metrics.PerfGetObjectTimeHistogram.WithLabelValues("get_object_get_data_time").Observe(time.Since(getDataTime).Seconds())
 }
 
-// tryDownloadAfterRecovery try to get piece data after data been recovery
-func (g *GateModular) tryDownloadAfterRecovery(ctx context.Context, pieceTask *gfsptask.GfSpDownloadPieceTask) ([]byte, error) {
+// tryDownloadAfterRecovery generate the recovery task on background, and try to get piece data after data been recovery
+// return recovered data and error
+func (g *GateModular) tryDownloadAfterRecovery(ctx context.Context, pieceTask *gfsptask.GfSpDownloadPieceTask, pInfo *downloader.SegmentPieceInfo, maxSegmentSize uint64) ([]byte, error) {
+	log.CtxDebugw(ctx, "recovery task key:", "key:", pInfo.SegmentPieceKey)
+	segmentIndex, parseErr := g.baseApp.PieceOp().ParseSegmentIdx(pInfo.SegmentPieceKey)
+	if parseErr != nil {
+		// no need to return recovery error to user
+		log.CtxErrorw(ctx, "fail to parse recovery segment index", "error", parseErr)
+		return nil, nil
+	}
+	segSize := g.baseApp.PieceOp().SegmentPieceSize(pieceTask.ObjectInfo.PayloadSize, segmentIndex, maxSegmentSize)
+	recoveryTask := &gfsptask.GfSpRecoverPieceTask{}
+	recoveryTask.InitRecoverPieceTask(pieceTask.GetObjectInfo(), pieceTask.GetStorageParams(),
+		coretask.DefaultLargerTaskPriority,
+		segmentIndex,
```

Comment:
these params can be moved to one line; don't exceed 120 columns.

Reply:
fixed